corehq/tests/test_classcleanup.py

```diff
@@ -0,0 +1,142 @@
+from functools import wraps
+from unittest import TestCase
+
+from nose.plugins import PluginTester
+from testil import eq
+
+from .noseplugins import classcleanup as mod
+
+
+class TestClassCleanupPlugin(PluginTester, TestCase):
+    activate = ''  # Activate option not needed. Plugin is always enabled.
+    plugins = [mod.ClassCleanupPlugin()]
+
+    def setUp(self):
+        pass  # super().setUp() is called by self.run_with_errors(...)
+
+    def makeSuite(self):
+        class Test(TestCase):
+            @classmethod
+            @maybe_error(self)
```

Comment:
I had no idea that `makeSuite` was a thing...and it's apparently [deprecated](https://github.com/python/cpython/blob/b50322d20337ca468f2070eedb051a16ee1eba94/Lib/unittest/loader.py#L488-L495) (and difficult to find documentation on).  Would it make sense, in each test, to construct this custom `Test` class manually and then execute it?

Reply:
I think you're thinking of `unittest.makeSuite`, which this is not. This is overriding [`PluginTester.makeSuite`](https://nose.readthedocs.io/en/latest/plugins/testing.html), which must be implemented if `self.suitepath` is `None`.